### ISSUE TITLE
support idl file

### ISSUE THREAD

**Issue by romanholidaypancakes** [beached/daw_json_link#324]
automatically output the corresponding structure according to idl,no need to manually enter the following code
```

namespace daw::json {
template <> 
struct json_data_contract<XXXXXXXXX> {
```

**Comment by beached** [comment]
This is the PR's are welcome part :).  I looked into trying to go from JSON Schema, it was inadequate to describe things like sum types/variants.  But it was also going to be a lot of work, it should be a separate project, and probably not something I can put the time into currently.  At this point, it looks like the IDL would need to be a new DSL.

However, I have a couple things that can help.
* In the v3, a lot of things like containers/numbers/string like types will generally work already
* With C++20 and v3, one can use Boost.Describe to generate the mappings for types in that system https://github.com/beached/daw_json_link_describe.  This is also similar to how it will work when C++ gets static reflection.
* https://github.com/beached/json_to_cpp can help generate the C++ data structures/mappings from JSON documents.  I used it in the past to generate the DS and mappings for arrays of JSON data.  It gets close.
